tests/functions.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+require '../functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class Functions extends TestCase
```

Comment:
Perhaps add success, failure and malformed tests for DisplayProducts.  Might be worth asking Charlie if you can get away without adding them though